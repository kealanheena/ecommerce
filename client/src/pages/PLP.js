import React, { useState } from 'react';
import { Container, Row } from 'react-bootstrap';

import productData from '../mock_data/products';
import ProductGrid from '../components/Product-grid';

function PLP() {
  const [products] = useState(productData);

  return (
    <Container className="plpPage">
      <Row>
        <h2 data-testid="PLP-header">PLP page</h2>
      </Row>
      <Row>
        <ProductGrid products={products} />
      </Row>
    </Container>
  );
}

export default PLP;
