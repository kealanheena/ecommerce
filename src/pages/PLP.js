import React, { useState } from 'react';

import productData from '../mock_data/products';

function PLP() {
  const [ products ] = useState(productData);

  return (
    <div className="plpPage">
      <h2>PLP page</h2>

      {products.map(({ title, price, description }) => (
        <div key={title}>
          <h3>{title}</h3>
          <p>{price}</p>
          <p>{description}</p>
        </div>
      ))}
    </div>
  );
}

export default PLP;
