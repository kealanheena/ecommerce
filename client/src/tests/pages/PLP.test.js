import React from 'react';
import { render } from '@testing-library/react';
import PLP from '../../pages/PLP';

test('Should render a <h2> with text of PLP page', () => {
  const { getByTestId } = render(<PLP />);
  expect(getByTestId('PLP-header')).toBeTruthy();
});
