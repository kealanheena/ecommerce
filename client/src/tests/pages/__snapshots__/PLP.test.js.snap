// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="plpPage container"
      >
        <div
          class="row"
        >
          <h2>
            PLP page
          </h2>
        </div>
        <div
          class="row"
        >
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Handcrafted Granite Cheese"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Handcrafted Granite Cheese
              </h5>
              <p>
                £717.00
              </p>
              <p>
                Handcrafted, Awesome, Small
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Tasty Metal Cheese"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Tasty Metal Cheese
              </h5>
              <p>
                £369.00
              </p>
              <p>
                Generic, Practical, Awesome
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Tasty Frozen Towels"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Tasty Frozen Towels
              </h5>
              <p>
                £900.00
              </p>
              <p>
                Small, Small, Licensed
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Tasty Steel Salad"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Tasty Steel Salad
              </h5>
              <p>
                £432.00
              </p>
              <p>
                Licensed, Sleek, Awesome
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Licensed Plastic Cheese"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Licensed Plastic Cheese
              </h5>
              <p>
                £445.00
              </p>
              <p>
                Sleek, Sleek, Refined
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Handcrafted Plastic Fish"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Handcrafted Plastic Fish
              </h5>
              <p>
                £340.00
              </p>
              <p>
                Unbranded, Fantastic, Sleek
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Licensed Steel Bacon"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Licensed Steel Bacon
              </h5>
              <p>
                £529.00
              </p>
              <p>
                Rustic, Intelligent, Licensed
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Tasty Steel Towels"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Tasty Steel Towels
              </h5>
              <p>
                £959.00
              </p>
              <p>
                Ergonomic, Handcrafted, Unbranded
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Ergonomic Soft Chicken"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Ergonomic Soft Chicken
              </h5>
              <p>
                £445.00
              </p>
              <p>
                Sleek, Gorgeous, Practical
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Generic Granite Chips"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Generic Granite Chips
              </h5>
              <p>
                £945.00
              </p>
              <p>
                Licensed, Incredible, Practical
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Licensed Cotton Car"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Licensed Cotton Car
              </h5>
              <p>
                £940.00
              </p>
              <p>
                Generic, Intelligent, Rustic
              </p>
            </div>
          </div>
          <div
            class="col-4"
          >
            <div
              class="product-card"
            >
              <img
                alt="Incredible Rubber Chicken"
                src="http://placehold.jp/640x480.png?text=placeholder"
                width="100%"
              />
              <h5>
                Incredible Rubber Chicken
              </h5>
              <p>
                £111.00
              </p>
              <p>
                Generic, Gorgeous, Tasty
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="plpPage container"
    >
      <div
        class="row"
      >
        <h2>
          PLP page
        </h2>
      </div>
      <div
        class="row"
      >
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Handcrafted Granite Cheese"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Handcrafted Granite Cheese
            </h5>
            <p>
              £717.00
            </p>
            <p>
              Handcrafted, Awesome, Small
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Tasty Metal Cheese"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Tasty Metal Cheese
            </h5>
            <p>
              £369.00
            </p>
            <p>
              Generic, Practical, Awesome
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Tasty Frozen Towels"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Tasty Frozen Towels
            </h5>
            <p>
              £900.00
            </p>
            <p>
              Small, Small, Licensed
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Tasty Steel Salad"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Tasty Steel Salad
            </h5>
            <p>
              £432.00
            </p>
            <p>
              Licensed, Sleek, Awesome
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Licensed Plastic Cheese"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Licensed Plastic Cheese
            </h5>
            <p>
              £445.00
            </p>
            <p>
              Sleek, Sleek, Refined
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Handcrafted Plastic Fish"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Handcrafted Plastic Fish
            </h5>
            <p>
              £340.00
            </p>
            <p>
              Unbranded, Fantastic, Sleek
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Licensed Steel Bacon"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Licensed Steel Bacon
            </h5>
            <p>
              £529.00
            </p>
            <p>
              Rustic, Intelligent, Licensed
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Tasty Steel Towels"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Tasty Steel Towels
            </h5>
            <p>
              £959.00
            </p>
            <p>
              Ergonomic, Handcrafted, Unbranded
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Ergonomic Soft Chicken"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Ergonomic Soft Chicken
            </h5>
            <p>
              £445.00
            </p>
            <p>
              Sleek, Gorgeous, Practical
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Generic Granite Chips"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Generic Granite Chips
            </h5>
            <p>
              £945.00
            </p>
            <p>
              Licensed, Incredible, Practical
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Licensed Cotton Car"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Licensed Cotton Car
            </h5>
            <p>
              £940.00
            </p>
            <p>
              Generic, Intelligent, Rustic
            </p>
          </div>
        </div>
        <div
          class="col-4"
        >
          <div
            class="product-card"
          >
            <img
              alt="Incredible Rubber Chicken"
              src="http://placehold.jp/640x480.png?text=placeholder"
              width="100%"
            />
            <h5>
              Incredible Rubber Chicken
            </h5>
            <p>
              £111.00
            </p>
            <p>
              Generic, Gorgeous, Tasty
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
